New meaning dict with changes from (HZB meeting) (#22)
SampleEnvironment/SECoP: protocol/specification/descriptive.rst

    Related issue: :issue:`008 Groups and Hierarchy`

.. _module-meaning:

``"meaning"``

>    A JSON object with data regarding the module meaning. It provides metadata that is useful for interpreting measurement data in an automatic fashion. It can have the keys ``function``, ``importance``, ``belongs_to``, ``link`` and ``key``, all of which are optional, with some restrictions. A meaning property can also be added on the :ref:`parameter-meaning <paremeter level>`.

Reviewer: type: "parameter level" instead of "parameter level"

Author: I am not sure what you mean here did you want to write "parameter-level" ?

I could also rewrite it to:
A meaning property can also be added as an optional parameter property.

Reviewer: sorry, my comment is confusing - its just a typo. you typed <paremeter level> instead of <parameter level>

Reviewer: again - its just a typo. you typed "paremeter" instead of "paramter"

    Related issue: :issue:`008 Groups and Hierarchy`

``"meaning"``

>    An unordered dictionary regarding the module meaning. It provides metadata that is useful for interpreting measurement data in a machine-readable format. It can have the keys ``function``, ``importance``, ``belongs_to``, ``link`` and ``key``, all of which are optional, with some restrictions. A meaning property can also be added on the :ref:`parameter-meaning <paremeter level>`.

Reviewer 2: ```suggestion
   A struct with data regarding the module meaning. It provides metadata that is useful for interpreting measurement data in an automatic fashion. It can have the keys ``function``, ``importance``, ``belongs_to``, ``link`` and ``key``, all of which are optional, with some restrictions. A meaning property can also be added on the :ref:`parameter-meaning <paremeter level>`.
```

Reviewer: I guess with the 'struct' you want to refer to the SECoP datainfo. However, we never said that SECoP datainfo may be applied to properties. Therefore it might be more correct to speak about a JSON object here.

Author: Yes, Ill change it to JSON object, I just looked and on other properties dicts/structs are named JSON objects as well.

   .. note::

>       In order for the meaning dicionary to be valid, it must contain at least a ``"link"`` or a ``"function"`` field.

Reviewer 2: ```suggestion
      In order for the meaning dictionary to be valid, it must contain at least a ``"link"`` or a ``"function"`` field.
```

   - ``"function"`` a string from an extensible list of predefined functions.

>       Predefined ``"functions"``:

Reviewer 2: This should be outdented by one space to be on the same level as the first backtick after the dash.

>          * ``"temperature"``

Reviewer 2: This needs an empty line between `Predefined functions` and the first item, and the bullets (asterisks) should be on the same level as `Predefined` to avoid a blockquote.

         * ``"temperature_regulation"`` (to be specified only if different from 'temperature')
         * ``"magneticfield"``
         * ``"electricfield"``
         * ``"pressure"``
         * ``"rotation_z"`` (counter clockwise when looked at 'from sky to earth')
         * ``"humidity"``
         * ``"viscosity"``
         * ``"flowrate"``
         * ``"concentration"``
         * ``"ph"``
         * ``"conductivity"``
         * ``"voltage"``
         * ``"surfacepressure"``
         * ``"stress"``
         * ``"strain"``
         * ``"shear"``
         * ``"heliumlevel"``

        This list may be extended later.

        ``_regulation`` may be postfixed, if the quantity generating module is different from the relevant measuring device. A regulation device MUST have an
        :ref:`interface class <interface-classes>` of at least ``Writable``.

        :related issue: :issue:`026 More Module Meanings`

>    - ``"importance"``  an integer value in the range ``[0,50]`` describing the importance. It allows ordering elements of the same tuple of ``"function"`` and ``"belongs_to"`` by importance.

Reviewer 2: ```suggestion
   - ``"importance"``  an integer value in the range ``[0,50]``. It allows ordering elements with the same tuple of ``"function"`` and ``"belongs_to"`` by importance.
```

      Predefined values:

>         * 10 means the instrument/beamline (Example: room temperature sensor always present)

Reviewer 2: same here, keep empty line before list

        * 20 means the surrounding sample environment (Example: VTI temperature)
        * 30 means an insert (Example: sample stick of dilution insert)
        * 40 means an addon added to an insert (Example: a device mounted inside a dilution insert)

      Intermediate values might be used. The range for each category starts at the indicated value minus 5 and ends below the indicated value plus 5.

      .. note::

>          - This field can only be present, if and only if there is an entry for ``"function"``

Reviewer 2: ```suggestion
         This field can only be present, if and only if there is an entry for ``"function"``.
```

      :related issue: :issue:`009 Module Meaning`

   - ``"belongs_to"`` a string identifying the entity to which the module is linked. Setting this field forms a relation between the entity and the ``"function"`` field.

      Predefined entities:

>          * ``"sample"``

Reviewer 2: empty line before and after list

         * ``"other"``
      .. note::
         - If not present, the default value ``"belongs_to":"other"`` is assumed.

>          - This field can only be present, if there is an entry for ``"function"``

Reviewer 2: ```suggestion
         - This field can only be present, if there is an entry for ``"function"``.
```

   - ``"link"`` a link to a vocabulary, glossary or ontology. Preferably a PID (Persistent Identifier) pointing to a specific entry.

   - ``"key"`` a key (string) that selects an entry from the knowledge representation that ``"link"`` points to. This mainly serves human readability if ``"link"`` already points to a specific entry.

      .. note::
         - This field must not be present if there is no ``"link"``
         - If ``"link"`` does not point directly to an entry, the ``"key"`` field is mandatory

    Example:

    .. code::

>     "meaning": {

Reviewer 2: code content needs to be indented

        "function": "temperature_regulation",
        "importance": 20,
        "belongs_to": "sample",
        "link": "https://w3id.org/nfdi4cat/voc4cat_0000051",
        "key": "synthesis temperature"
    }

    This reads as:

>     Regulation of the sample (``belongs_to``) temperature (``function``) in the surrounding sample environment (``importance``).The ``key`` and ``link`` give additional metadata, saying that the regulated temperature is also the ``synthesis temperature`` of the experiment.

Reviewer 2: ```suggestion
    Regulation of the sample (``belongs_to``) temperature (``function``) in the surrounding sample environment (``importance``) .The ``key`` and ``link`` give additional metadata, saying that the regulated temperature is also the ``synthesis temperature`` of the experiment.
```

    The value given here must conform to the data type of the accessible.

.. _parameter-meaning:
``"meaning"``

>    A dictionary regarding the parameter meaning. It has the same specification as the :ref:`module-meaning <module meaning>`.

Reviewer 2: Is this reference defined?

Author: ah thanks i think this one got lost when i recreated the branch
